### PR TITLE
Ensure that version used my plugin loader has the correct syntax

### PR DESCRIPTION
**Pull request by sredxny**
## Description

When reading the version we need to ensure that it contains the prefix "v". It happens that goreleaser strips the prefix, so we have different behaviors when building the app locally vs the container built by goreleaser. For more details: [documentation](https://goreleaser.com/customization/templates/?h=variables#fn:version-prefix) 

## Related Issue

https://tyktech.atlassian.net/browse/TT-6668

## Motivation and Context

Problem details: https://tyktech.atlassian.net/browse/TT-6668

## How This Has Been Tested

- Run the next steps while building the app locally and also running the gw using an RC image.

- Create an api via dashboard with a goplugin enabled. The plugin file doesn't really have to exist, as we need to check is the file that attempts to load. In order to do so, I edited the apidefinition in the raw editor and added:

```
"custom_middleware": {
      "pre": [
          {
            "name": "AddFooBarHeader",
            "path": "anypath/plugin.so"
        }
    ],
      "driver": "goplugin",
     // the other stuffs
}
```
- Run the gw and check the name of the plugin that is attempting to load...the version section should be in the format: vx.x.x

In my local I get the next log line, which shows me that the plugin loader is targeting the correct file:

```
ERROR Could not load Go-plugin error=plugin.Open("anypath/plugin_v4.1.0_darwin_amd64.so"): realpath failed mwPath=anypath/plugin.so mwSymbolName=AddFooBarHeader
```

To use docker containers I generated the image `tykio/tyk-gateway:v4.3.0-rc5` that contain the changes in this branch, however I encourage you to create your own image.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning why it's required
- [ ] I would like a code coverage CI quality gate exception and have explained why

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4417/merge
Commit:  
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3535218428)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4417/merge
Commit: df8d2f43d3989128e642425daaccf2d911c2496b 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3537247115)

**Review by tbuchaillot — APPROVED** [review]
LGTM

File: gateway/mw_go_plugin.go [hunk 1]
```diff
@@ -255,9 +255,18 @@ func (m *GoPluginMiddleware) goPluginFromTykVersion(version string) string {
...
 
 	newPluginName := strings.Join([]string{pluginName, version, os, architecture}, "_")
 	newPluginPath := pluginDir + "/" + newPluginName + ".so"
 
 	return newPluginPath
 }
+
+// EnsureSemanticVersioning receives a version and check that it has the prefix 'v' otherwise, it adds it
+func EnsureSemanticVersioning(version string) string {
+	if !strings.HasPrefix(version, "v") {
```
Comment (mativm02) [inline gateway/mw_go_plugin.go:268]: Hey @sredxny, what do you think about adding a second conditional to check if the letter "v" is in the first place of the string? I mean, the version could be "1.3.3v" and the function wouldn't detect it. Btw, I don't know if that's something possible to happen, just asking :)

Comment (sredxny) [reply gateway/mw_go_plugin.go:268]: @mativm02 that is exactly what `strings.HasPrefix` does. It checks that the string starts with "v"

Comment (mativm02) [reply gateway/mw_go_plugin.go:268]: @sredxny I don’t know why I thought you were using strings.Contains(). Sorry for the confusion.

**Review by mativm02 — APPROVED** [review]
lgtm!

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4417/merge
Commit: b08dc01c6b07ce9efbaa6d451ce3439cb9979fa7 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3544086702)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4417/merge
Commit: a6be0411a1fdc6afc7500115b8e2d1dc0dce8a39 
Triggered by: pull_request (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3550842006)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: 3339c4a67655210f1d92dc4e466e6c6620d9f832 Ensure that version used my plugin loader has the correct syntax (#4417)

* ensure that version used my plugin loader has the correct syntax
Triggered by: push (@sredxny)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3550881408)

**Pull request merged**